### PR TITLE
Add example of styled-components usage

### LINKED ISSUE #190: Check compatibility with Styled components
https://www.styled-components.com/

### PR DESCRIPTION
**Pull request by awinogradov**
Close #190 

сс @rakchaev

### CODE REVIEW

File: examples/styled-components/README.md [hunk 1]
```diff
@@ -0,0 +1,2229 @@
+This project was bootstrapped with [Create React App](https://github.com/facebookincubator/create-react-app).
```
Comment (Yeti-or) [inline examples/styled-components/README.md]: Let's describe example here
how to build it
and what it uses inside

File: examples/styled-components/src/components/Example/Example.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+import { decl } from 'bem-react-core';
+import styled from 'styled-components';
+
+import addStyle from '../../styleHOC';
+
+const style = Component => styled(Component)`
```
Comment (Yeti-or) [inline examples/styled-components/src/components/Example/Example.js]: By the way will severSiderRendering work with such declaration?

Comment (awinogradov) [reply examples/styled-components/src/components/Example/Example.js]: It's example on create-react-app, it doesn't work with ssr.

File: examples/styled-components/README.md (hunk 1 above)
Comment (awinogradov) [reply examples/styled-components/README.md]: Instruction in the readme is correct. `npm i && npm start`

Comment (Yeti-or) [reply examples/styled-components/README.md]: I mean it huge Readme
there is no need of it
we need to describe what is special in these particular example

**Pull request merged**